File: javascript/reverse-string/reverse-string.js
//
// This is only a SKELETON file for the 'Reverse String' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const reverseString = (wordToReverse) => {
  let charArray = wordToReverse.split("");
  let reversedString = [];
  for (var i = charArray.length; i > - 1; i--) {
    reversedString.push(charArray[i]);
  }

  return reversedString.join("");
};

File: cpp/reverse-string/reverse_string.cpp
#include "reverse_string.h"

using namespace std;

namespace reverse_string {

  string reverse_string(string word)
  {
    string reverse;
    for (int i = word.length() - 1; i > -1; i--)
    {
      reverse += word[i];
    }

    return reverse;
  }

}  // namespace reverse_string
